gui-backend/src/main/java/com/rln/gui/backend/implementation/methods/AutoapproveApiImpl.java

```diff
@@ -56,6 +67,42 @@ public void updateApprovalProperties(@Valid @NotNull ApprovalProperties autoAppr
     rlnClient.createOrUpdateAutoApproveMarker(parameters);
   }
 
+  public List<LedgerAddressDTO> get() {
+    var accounts = accountCache.getAccounts();
+    ArrayList<LedgerAddressDTO> result = new ArrayList<>(accounts.size());
+
+    for (var account : accounts) {
+      var autoApproval = autoApproveCache.getMarker(account.getKey());
+      result.add(LedgerAddressDTO.builder()
+          .isIBAN(true)
+          .address(account.getKey())
+          .bearerToken("")
+          .clientId(0L)
+          .approvalMode(convertApproveType(autoApproval.data.autoApproveType))
+          .approvalLimit(getLimit(autoApproval.data.autoApproveType))
+          .build());
+    }
+
+    return result;
+  }
+
+  private Double getLimit(AutoApproveType autoApproveType) {
+    if (autoApproveType instanceof LimitedMaxAmount) {
+      return ((LimitedMaxAmount) autoApproveType).bigDecimalValue.doubleValue();
+    }
+    return Double.NaN;
```

Comment:
This is interesting. I'd also consider `null` here, we don't have a better answer. Would be good to discuss this offline.